### PR TITLE
Added option to show/hide secrets in status output

### PR DESCRIPTION
**Pull request by nik-redhat**
### Explain the changes
Masked the ACCESS_KEYS and passwords to * by default, in status output.
And, if the `--show-secrets` flag is passed to the command the secrets
are shown.

Fixes: #928

### Testing Instructions:
1. Try running the command `noobaa status` after successful installation, you should see the secrets as `*`
2. Run the same command, but this time with the `--show-secrets` flag. The secrets will be visible.

Signed-off-by: nik-redhat <nladha@redhat.com>

### CODE REVIEW

**Comment by liranmauda** [comment]
Hi @nik-redhat 
Thanks for the PR 😄 

I have run the tests. I believe they will fail as they depend on the credentials, So I guess  the tests also should be fixed and call using this flag.
One place I can think of is:
https://github.com/noobaa/noobaa-operator/blob/f325c18cfa94b33d54c8e8b05a5edf7ad962615a/test/cli/test_cli_functions.sh#L697-L707

@dannyzaken @kfir-payne will it make sense to use `maskString` here?

**Comment by nik-redhat** [comment]
> I have run the tests. I believe they will fail as they depend on the credentials, So I guess the tests also should be fixed and call using this flag. One place I can think of is:
> 
> https://github.com/noobaa/noobaa-operator/blob/f325c18cfa94b33d54c8e8b05a5edf7ad962615a/test/cli/test_cli_functions.sh#L697-L707
> 

Yes, forgot about the tests. There are a couple of other places too, that might need an update.
Thanks for the pointer :)

> @dannyzaken @kfir-payne will it make sense to use `maskString` here?

I was searching for something like this (as a pkg), and it turns out we have some masking type already in place. Hopefully, we should be able to use `maskString` here, and it will also keep the logic consistent, throughout the codebase.

File: pkg/system/system.go [hunk 1]
```diff
@@ -577,7 +579,12 @@ func RunStatus(cmd *cobra.Command, args []string) {
 	util.KubeCheck(secret)
 
 	fmt.Println("")
-	fmt.Printf("%s password : %s\n", secret.StringData["email"], secret.StringData["password"])
+	if options.ShowSecrets {
+		fmt.Printf("%s password : %s\n", secret.StringData["email"], secret.StringData["password"])
+	} else {
+		fmt.Printf("%s password : %s\n", secret.StringData["email"],
+			re.ReplaceAllString(secret.StringData["password"], "*"))
```
Comment (dannyzaken) [inline pkg/system/system.go:586]: as @liranmauda commented, you can use [MaskedString](https://github.com/noobaa/noobaa-operator/blob/f325c18cfa94b33d54c8e8b05a5edf7ad962615a/pkg/nb/types.go#L18-L22)

```suggestion
		fmt.Printf("%s password : %s\n", secret.StringData["email"], nb.MaskedString(secret.StringData["password"]))
```

**Comment by nik-redhat** [comment]
Updated the code to use `MaskedString`.

**Comment by dannyzaken** [comment]
@nik-redhat can you squash your commits?

**Review by liranmauda — APPROVED** [review]
LGTM

File: pkg/obc/obc.go [hunk 2]
```diff
@@ -383,8 +383,13 @@ func RunStatus(cmd *cobra.Command, args []string) {
...
 	for k, v := range secret.StringData {
 		if v != "" {
-			fmt.Printf("  %-22s : %s\n", k, v)
-			credsEnv += k + "=" + v + " "
+			if options.ShowSecrets {
+				fmt.Printf("  %-22s : %s\n", k, v)
+				credsEnv += k + "=" + v + " "
+			} else {
+				fmt.Printf("  %-22s : %s\n", k, nb.MaskedString(v))
+				credsEnv += k + "=********* " // Can't use nb.MaskedString() here because of type mismatch
```
[the remark this answers is not in the document]
Comment (nik-redhat) [reply pkg/obc/obc.go]: Oh, right.
Fixed it now.

**Review by dannyzaken — APPROVED** [review]

**Comment by dannyzaken** [comment]
@nik-redhat LGTM. give us an ack and we'll merge

**Comment by nik-redhat** [comment]
> @nik-redhat LGTM. give us an ack and we'll merge

All good from my end.
Please merge at will. Thank you :)

**Pull request merged**